test/enum_test.js

```diff
@@ -0,0 +1,132 @@
+import { expect } from "chai";
+
+import { createFormComponent, createSandbox } from "./test_utils";
+
+describe("enum", () => {
+  let sandbox;
+
+  beforeEach(() => {
+    sandbox = createSandbox();
+  });
+
+  afterEach(() => {
+    sandbox.restore();
+  });
+
+  it("should render a select element if set the enum.", () => {
+    const schema = {
+      type: "object",
+      properties: {
+        foo: {
+          type: "string",
+          enum: [0],
+        },
+      },
+    };
+
+    const { node } = createFormComponent({
+      schema,
+    });
+
+    const selects = node.querySelectorAll("select");
+    expect(selects).to.have.length.of(1);
+  });
+
+  it("should render a select element and it's value is empty, if set the enum and the default value is undefined.", () => {
+    const schema = {
+      type: "object",
+      properties: {
+        foo: {
+          type: "string",
+          enum: [0],
+        },
+      },
+    };
+
+    const { node } = createFormComponent({
+      schema,
+    });
+
+    const selects = node.querySelectorAll("select");
+    expect(selects[0].value).eql("");
+  });
+
+  it("should render a select element and it's first option has an empty innerHTML, if set the enum and the default value is undefined.", () => {
+    const schema = {
+      type: "object",
+      properties: {
+        foo: {
+          type: "string",
+          enum: [0],
+        },
+      },
+    };
+
+    const { node } = createFormComponent({
+      schema,
+    });
+
+    const options = node.querySelectorAll("option");
+    expect(options[0].innerHTML).eql("");
+  });
+
+  it("should render a select element and it's first option is '0', if set the enum and the default value is 0.", () => {
+    const schema = {
+      type: "object",
+      properties: {
+        foo: {
+          type: "string",
+          enum: [0],
+          default: 0,
+        },
+      },
+    };
+
+    const { node } = createFormComponent({
+      schema,
+    });
+
+    const options = node.querySelectorAll("option");
+    expect(options[0].innerHTML).eql("0");
+  });
+
+  it("should render a select element and it's first option is 'false', if set the enum and the default value is false.", () => {
```

Comment:
Can you also add a test in which the default value is '', and then it shouldn't render two empty options?

Reply:
Thank you for your suggestions, i have adjusted it.